database/models/Site.ts

```diff
@@ -22,23 +22,88 @@ export class Site extends Model {
   id!: number
 
   @Column({
+    unique: true,
     allowNull: false,
     type: DataType.TEXT,
   })
-  name!: string
+  repositoryName!: string
 
   @Column({
-    allowNull: false,
     type: DataType.TEXT,
   })
-  apiTokenName!: string
+  createdBy!: string | null
 
   @CreatedAt
   createdAt!: Date
 
   @UpdatedAt
   updatedAt!: Date
 
+  @Column({
+    type: DataType.TEXT,
+  })
+  agency!: string | null
```

Comment:
this annotation is wrong - if we assert that this is strictly non-null, the type should jsut be `string`

Reply:
My style is to use database constraints sparingly...only when failing to observe them would be catastrophic. Policies about how data should be gathered and structured tend to change over time, and if policies are built into database constraints, then changing them may require painfully complicated database operations. In most cases, I think it's better to build constraints into code, because code can be changed very easily. In this case, that means 

1. putting most constraints into the Joi validators, so requests with bad data will be rejected.
2. making as few assumptions as possible when reading the data, in case it's "dirty"

Reply:
you're right! turns out they use the same exclamation mark ._. just that it's different based on context

however, this is still an issue on our data model - we distinguish sites based on whether they are launched or not, and our data model should reflect that. see [here](https://fsharpforfunandprofit.com/series/designing-with-types/) (in f# but idea is same).

similar to the above, we should aim to have 3 kinds - `Site, LaunchedSite, UnlaunchedSite` where `UnlaunchedSite (and LaunchedSite) extends Site` and tags on extra meta data. this data model fits our reality well and is extensible in the future.